File: API/WebHeroAPI/Models/Song.cs
namespace WebHeroAPI.Models
{
    public class Song
    {
        public string? Name { get; set; }
        public string? Artist { get; set; }
        public string? Year { get; set; }
        public string? Album { get; set; }
        public int? Length { get; set; }
        public int? NoteCount { get; set; }
        public int? NPS { get; set; }
        public int? Difficulty { get; set; }
        public string? ImageURL { get; set; }
        public string? Hash { get; set; }
    }
}

File: Client/WebHeroClient/src/app/models/song.ts
export class Song {
    constructor(
        public name: string,
        public artist: string,
        public year: string,
        public album: string,
        public length: number, // seconds
        public noteCount: number,
        public NPS: number,
        public difficulty: number, // out of 7
        public imageURL: string,
        public hash: string
    ){}
}